app/celery/tasks/agent_tasks.py

```
@@ -185,89 +200,102 @@ async def run_regeneration():
-                )
 
             # Ensure the database session is properly committed after regeneration
             # The flush_message_buffer should have been called at the end of the generator
             # but we need to ensure the session is committed
             try:
-                self.db.commit()
+                await async_db.commit()
                 logger.info(
                     f"Database session committed after regeneration for conversation {conversation_id}"
                 )
             except Exception as commit_error:
```

Ruff (0.14.0)

Do not catch blind exception: `Exception`

(BLE001)